src/sbt-test/sbt-scalafix/resolvers/test

```diff
@@ -0,0 +1,7 @@
+# without the proper resolver/repository, invocation of the community rule fails
+-> scalafix SyntacticRule
+
+# once the proper resolver is setup via sbt, dependency fetching works fine and scalafix can be used normally
+> set ThisBuild / resolvers ++= Resolver.sonatypeOssRepos("snapshots")
+> scalafix SyntacticRule
```

Comment:
agree, fixed

Reply:
but you removed the last `> scalafix --check SyntactRule` ? you should keep it since that's the one really showing scalafix had an impact

Reply:
Uh, I thought bcs we haven't actually put the expected file into dir, so the --check would fail since the expected fixed is empty. did I missed something?

Reply:
this is not an expect test, it's just a raw, mutable file for which the SyntacticRule rule needs to apply a modification, so before the scalafix invocation, `--check` should fail, and after it should suceeed.

Reply:
```
> set ThisBuild / resolvers ++= Resolver.sonatypeOssRepos("snapshots")
-> scalafix --check SyntacticRule
> scalafix --check SyntacticRule
> scalafix SyntacticRule
```
I added this missing one and the test failed after that, so is this means there's something wrong with implementation? I'm investigating into it.

Reply:
it might be related to https://github.com/scalacenter/sbt-scalafix/pull/404#discussion_r1568499634

Reply:
ah no, you just didn't get the order right, it should be (with comments just to explain why I was advocating for a double --check)

```
> set ThisBuild / resolvers ++= Resolver.sonatypeOssRepos("snapshots")
-> scalafix --check SyntacticRule # make sure we have a failure despite having the resolver
> scalafix SyntacticRule
> scalafix --check SyntacticRule # confirm that scalafix works since the command that didn't work before now does
```

Reply:
Uh, I see, thanks